tap_linkedin_ads/sync.py

```diff
@@ -518,3 +608,159 @@ def sync(client, config, catalog, state):
+        'start': start,
+        'count': count,
+        **static_params,
+        'fields': ','.join(chunk),
+    }
+    if bookmark_query_field:
+        params[bookmark_query_field] = last_datetime
+
+    querystring = '&'.join(['%s=%s' % (key, value) for (key, value) in params.items()])
+    next_url = 'https://api.linkedin.com/v2/{}?{}'.format(path, querystring)
+
+    while next_url:
+        LOGGER.info('ANALYTICS URL for %s: %s', stream_name, next_url)
+
+        # Get data, API request
+        data = client.get(
+            url=next_url,
+            endpoint=stream_name)
+        yield data
+
+        total_records += len(data['elements'])
+
+        # Pagination: Get next_url
+        next_url = None
+        links = data.get('paging', {}).get('links', [])
+        for link in links:
+            rel = link.get('rel')
+            if rel == 'next':
+                href = link.get('href')
+                if href:
+                    next_url = 'https://api.linkedin.com{}'.format(urllib.parse.unquote(href))
+
+        LOGGER.info('%s: Synced page %s, this page: %s. Total records processed: %s',
+                    stream_name,
+                    page,
+                    len(data['elements']),
+                    total_records)
+        page = page + 1
+
+
+def sync_ad_analytics_by_campaign(client, #pylint: disable=too-many-branches,too-many-statements
+                                  catalog,
+                                  state,
+                                  start_date,
+                                  stream_name,
+                                  path,
+                                  endpoint_config,
+                                  data_key,
+                                  static_params,
+                                  bookmark_query_field=None,
+                                  bookmark_field=None,
+                                  parent=None,
+                                  parent_id=None):
+
+    write_schema(catalog, stream_name)
+
+    # Get the latest bookmark for the stream and set the last_datetime
+    last_datetime = get_bookmark(state, stream_name, start_date)
+    max_bookmark_value = last_datetime
+    LOGGER.info('%s: bookmark last_datetime = %s', stream_name, max_bookmark_value)
+
+    total_records = 0
+    start = 0
+    count = endpoint_config.get('count', 100)
+    valid_selected_fields = [snake_case_to_camel_case(field)
+                             for field in selected_fields(catalog.get_stream(stream_name))
+                             if snake_case_to_camel_case(field) in FIELDS_AVAILABLE_FOR_AD_ANALYTICS_V2]
+
+    # chunk size is one less than the max number of metrics (20) to make room for dateRange
+    # If the schema changes and there's more than 76 fields, this has to be revisited
+    chunkA = valid_selected_fields[0:19]
+    chunkB = valid_selected_fields[19:38]
+    chunkC = valid_selected_fields[38:57]
+    chunkD = valid_selected_fields[57:76]
```

Comment:
I think this should be a generator that pops off 19 + dateRange. It would be more clear.